File: frameworks/ruby-rails-fixed/app/graphql/types/post_type.rb
module Types
  class PostType < Types::BaseObject
    field :id, ID, null: false
    field :title, String, null: false
    field :content, String, null: true
    field :created_at, GraphQL::Types::ISO8601DateTime, null: false

    field :author, Types::UserType, null: false
    field :comments, [Types::CommentType], null: false

    def author
      User.find_by(pk_user: object.fk_author)
    end

    def comments
      Comment.where(fk_post: object.pk_post)
    end
  end
end

File: frameworks/csharp-dotnet/FraiseQL.Benchmark/GraphQL/Types/PostType.cs
using FraiseQL.Benchmark.GraphQL.Resolvers;
using FraiseQL.Benchmark.Models;

namespace FraiseQL.Benchmark.GraphQL.Types;

public class PostType : ObjectType<Post>
{
    protected override void Configure(IObjectTypeDescriptor<Post> descriptor)
    {
        descriptor.Field(p => p.Id);
        descriptor.Field(p => p.Title);
        descriptor.Field(p => p.Content);
        descriptor.Field(p => p.FkAuthor).Name("authorId");
        descriptor.Field(p => p.CreatedAt);
    }
}